src/processes/helper/sendTopicAutoMessage.js

```diff
@@ -0,0 +1,61 @@
+const momentTz = require('moment-timezone');
+const {v4: uuidv4} = require('uuid');
+const {sample} = require('lodash');
+const {followTopicQueue} = require('../../config/redis');
+const {CommunityMessageFormat, LogError} = require('../../databases/models');
+
+const followTopicServiceQueue = async (userId, topicId, communityMessageFormatId, delay) => {
+  const data = {
+    userId,
+    topicId,
+    communityMessageFormatId
+  };
+  const currentTime = momentTz().tz('America/Los_Angeles');
+  const randomTime = sample([6, 7, 8]);
+  const additionalDays = delay;
+
+  const requiredTime = momentTz()
+    .tz('America/Los_Angeles')
+    .set({hour: randomTime})
+    .add(additionalDays, 'days');
+
+  const diffTime = requiredTime.diff(currentTime, 'milliseconds');
+
+  const options = {
+    jobId: uuidv4(),
+    removeOnComplete: true,
+    delay: diffTime
+  };
+  try {
+    await followTopicQueue.add(data, options);
+  } catch (e) {
+    console.log('error', e);
+  }
+};
+const sendTopicAutoMessage = async (userId, topicIds) => {
+  try {
+    topicIds.forEach(async (topicId) => {
+      const communityMessageFormats = await CommunityMessageFormat.findAll({
+        where: {topic_id: topicId, status: '1'}
+      });
+
+      if (communityMessageFormats.length > 0) {
+        communityMessageFormats.forEach(async (communityMessageFormat) => {
+          followTopicServiceQueue(
+            userId,
+            topicId,
+            communityMessageFormat.community_message_format_id,
+            communityMessageFormat.delay
+          );
+        });
+      }
+    });
+  } catch (error) {
+    await LogError.create({
+      message: error.message
+    });
+    console.error('Register send Topic auto message: ', error);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error message is being logged to the console but it would be better to include more context about the error. Consider including the user ID and topic IDs in the error message.

```diff
- console.error('Register send Topic auto message: ', error);
+ console.error(`Register send Topic auto message for user ${userId} with topics ${topicIds.join(', ')}: `, error);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->